velox/functions/prestosql/tests/ComparisonsTest.cpp

```diff
@@ -97,6 +97,60 @@ TEST_F(ComparisonsTest, betweenDate) {
   }
 }
 
+TEST_F(ComparisonsTest, betweenDecimal) {
+  this->options_.parseDecimalAsDouble = false;
+  auto runAndCompare = [&](const std::string& exprStr,
+                           VectorPtr input,
+                           std::vector<bool>& expectedResult) {
+    auto actual = evaluate<SimpleVector<bool>>(exprStr, makeRowVector({input}));
+    for (auto i = 0; i < expectedResult.size(); ++i) {
+      EXPECT_EQ(actual->valueAt(i), expectedResult[i]);
+    }
+  };
+  auto shortFlat = makeNullableShortDecimalFlatVector(
+      {100, 250, 300, 500, std::nullopt}, DECIMAL(3, 2));
+  std::vector<bool> expectedResult{false, true, true, false, false};
+
+  runAndCompare("c0 between 2.00 and 3.00", shortFlat, expectedResult);
+
+  auto longFlat = makeNullableLongDecimalFlatVector(
+      {100, 250, 300, 500, std::nullopt}, DECIMAL(20, 2));
+
+  // Comparing LONG_DECIMAL and SHORT_DECIMAL must throw error.
+  VELOX_ASSERT_THROW(
+      runAndCompare("c0 between 2.00 and 3.00", longFlat, expectedResult),
+      "Scalar function signature is not supported: between(LONG_DECIMAL(20,2), SHORT_DECIMAL(3,2), SHORT_DECIMAL(3,2)).");
+}
+
+TEST_F(ComparisonsTest, eqDecimal) {
+  auto runAndCompare = [&](const std::string& expr,
+                           std::vector<VectorPtr>& inputs,
+                           std::vector<bool>& expectedResult) {
+    auto actual =
+        evaluate<SimpleVector<bool>>("c0 == c1", makeRowVector(inputs));
+    for (auto i = 0; i < expectedResult.size(); ++i) {
+      EXPECT_EQ(actual->valueAt(i), expectedResult[i]);
+    }
+  };
+
+  std::vector<VectorPtr> inputs = {
+      makeNullableShortDecimalFlatVector(
+          {1, std::nullopt, 3, -3, std::nullopt, 4}, DECIMAL(10, 5)),
+      makeNullableShortDecimalFlatVector(
+          {1, 2, 3, -3, std::nullopt, 5}, DECIMAL(10, 5))};
+  auto expected = std::vector<bool>{true, false, true, true, false, false};
```

Comment:
When comparing with NULL, the result is NULL. Hence, the expected result should be NULL for rows 1 and 4.

```
> select 1.4 = null;
 _col0
-------
 NULL
(1 row)
```